s3/pkg/datastore/ceph/ceph.go

```diff
@@ -319,7 +336,26 @@ func (ad *CephAdapter) ListParts(ctx context.Context, multipartUpload *pb.ListPa
 }
 
 func (ad *CephAdapter) BackendCheck(ctx context.Context, backendDetail *pb.BackendDetailS3) error {
-    return ErrNotImplemented
+	randId := uuid.NewV4().String()
+	input := &pb.Bucket{
+		Name: sampleBucket + randId,
+	}
+
+	err := ad.BucketCreate(ctx, input)
```

Comment:
Is BucketCreate synchronous call? 
If it's async, immediate bucketDelete may through error

Reply:
ok @kumarashit will check and update the code

Reply:
@kumarashit I have updated the code and handled the issue by calling Head bucket